src/writer.rs

```diff
@@ -85,7 +115,19 @@ impl<W: Write> Writer<W> {
             Event::PI(ref e) => self.write_wrapped(b"<?", e, b"?>"),
             Event::DocType(ref e) => self.write_wrapped(b"<!DOCTYPE", e, b">"),
             Event::Eof => Ok(0),
+        };
+        if self.indent_size > 0 {
+            match *event.as_ref() {
+                Event::Start(_) => {
+                    let new_len = self.indents.len() + self.indent_size;
+                    self.indents.resize(new_len, self.indent_char);
```

Comment:
As explained before, the `resize` can be done only if the indent length is not sufficient.